fitsio/fitslib.py

```diff
@@ -3692,8 +3700,18 @@ def __getitem__(self, arg):
             self.fitshdu._process_args_as_rows_or_columns(arg, unpack=True)
         if isrows:
             # rows was entered: read all current column subset
-            return self.read(rows=res)
-
+            r = self.read(rows=res)
+            if numpy.isscalar(res):
```

Comment:
This code looks wrong anyways. See the documentation of numpy.isscalar, https://docs.scipy.org/doc/numpy/reference/generated/numpy.isscalar.html

Reply:
Well, it is detecting the input is a scalar but the warning doesn't happen.

Reply:
The test here needs to be on how the user has indexed the array. In other words, we need to know if they passed something like '0:1` or `0`. The output is always a vector right now and so this if condition will always evaluate to `False`.

Reply:
We cannot test for a length 1 item since this would not properly distinguish the two cases either.

Reply:
that test is on `res` which is the rows, not the output `r`.

I agree we might want some more sophisticated handling here.

Reply:
Ack. OK. This PR needs work.

Reply:
Do we need a test for a tuple to catch whole rows?

Reply:
Wait no that won't work.

Reply:
Changed to use ndim instead. I didn't know this!